### PR TITLE
UI: better encapsulation, tests and some cleanup for Triggerer-trait

### PR DESCRIPTION
**Pull request by klees**
As discovered [here](https://github.com/ILIAS-eLearning/ILIAS/pull/1098), the trait `Triggerer` deserves some better encapsulation, since the current form leads to hard to understand code. I also added tests for it and cleaned up a method name to match naming conventions.

### CODE REVIEW

File: src/UI/Implementation/Component/Triggerer.php [hunk 1]
```diff
@@ -65,12 +65,28 @@ protected function setTriggeredSignal(Component\Signal $signal, $event) {
...
+	 * @param	string
+	 * @return \ILIAS\UI\Component\Signal
+	 */
+	public function getTriggeredSignalsFor($event) {
+		if (!isset($this->triggered_signals[$event])) {
+			return [];
+		}
+		return array_map(
+			function($ts) { return $ts->getSignal(); },
+			$this->triggered_signals["click"]
```
Comment (chfsx) [inline src/UI/Implementation/Component/Triggerer.php]: This should be `$this->triggered_signals[$event]` as well, right?

File: src/UI/CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # 5.2.4
 
-* ButtonFactory::standard, ButtonFactory::primary, ButtonFactory::shy, ButtonFactory::tag
+* `ButtonFactory::standard`, `ButtonFactory::primary`, `ButtonFactory::shy`, `ButtonFactory::tag`
   can get a signal as second parameter instead of an action. If an url was set, the
   previously defined action-url will be deleted when a click-signal is added. Button::getAction
   may thus also return a list of Signals instead of an url-string.
-
+* `ILIAS\UI\Implementation\Component\Triggerer::addTriggeredSignal` has been renamed to
```
Comment (chfsx) [inline src/UI/CHANGELOG.md:7]: old and new name seem to be identical.

File: src/UI/Implementation/Component/Triggerer.php (hunk 1 above)
Comment (klees) [reply src/UI/Implementation/Component/Triggerer.php]: Yes. And the test should find it. Good catch, thx!

**Review by Amstutz — APPROVED** [review]
LGTM. There are some Interface changes. Do we discuss this in the next JF?

**Comment by matthiaskunkel** [comment]
Jour Fixe, 27 Aug 2018 : Richard notified the JF about a change in the public interface.

**Comment by Amstutz** [comment]
How do we proceed with this? Is this ready to be merge? Is there an interference with https://github.com/ILIAS-eLearning/ILIAS/pull/1098? If so, how do we prevent to not make the requirements for https://github.com/ILIAS-eLearning/ILIAS/pull/1098 higher?

**Comment by klees** [comment]
IMO this is ready to be merged. As far as I can see this is also already included in #1098.

**Pull request merged**